### DISCUSSION TITLE
OCaml syntax highlighting doesn't escape labelled strings

### DISCUSSION THREAD

**Discussion by zbaylin** [community/community#42534] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

The OCaml syntax highlighter breaks when using labelled strings, i.e. `{label|my_str|label}`.

Example:

```ocaml
let x = {foo| bar (* |foo} in

let my_int = 10
```

The `(*` would ordinarily start a comment, but should not since it's in a string. This works correctly if the string isn't labelled:

```ocaml
let x = " bar (* "

let my_int = 10
```

**Comment by frectonz** [comment]
It still doesn't work :cry: 

```ocaml
let x = {|
hello"
this
is
multi-line 
;; \
(*
|}

let a = 10
let b = "hello"
```

This is what it is supposed to look like
![image](https://github.com/community/community/assets/53809656/21154488-9e50-44ed-96e9-5a743b779cc1)
